src/main/features/claimant-response/views/defendants-response.njk

```diff
@@ -0,0 +1,167 @@
+              paymentPlan = paymentPlan
+            )
+          }}
+        {% endif %}
+
+        {{ paymentDateExplanationBlock(claim.response.statementOfMeans.reason) }}
+
+        {{ financialDetailsBlock(totalMonthlyIncome, totalMonthlyExpenses, claim.response.statementOfMeans) }}
+      {% endif %}
+
+      {% if claim.response.responseType === ResponseType.PART_ADMISSION.value %}
+        {% if page === 0 %}
+          <p>{{ t('{{ defendantName }} admits they owe you {{ amount }}. However they don’t believe they owe the full amount claimed.', { defendantName: defendantName, amount: claim.response.amount | numeral }) }}</p>
+          <p>
+            {{
+              paymentIntentionFragment(
+                amount = undefined,
+                paymentIntention = claim.response.paymentIntention
+              )
+            }}
+          </p>
+
+          <h2 class="heading-medium">{{ t('Their defence') }}</h2>
+
+          <h3 class="heading-small form-group-related">{{ t('Why they don’t owe the amount claimed') }}</h3>
+          <p>{{ t('"{{ defence }}"', { defence: claim.response.defence }) }}</p>
+
+          {% if claim.response.timeline and claim.response.timeline.rows.length > 0 %}
+            <h3 class="heading-small form-group-related">{{ t('Their timeline of events') }}</h3>
+            {{ timelineFragment(claim.response.timeline) }}
+          {% endif %}
+
+          {% if claim.response.evidence and claim.response.evidence.comment %}
+            <h3 class="heading-small form-group-related">{{ t('They disagree with your evidence') }}</h3>
+            <p>{{ t('"{{ comment }}"', { comment: claim.response.evidence.comment }) }}</p>
+          {% endif %}
+        {% elseif page === 1 %}
+          {% if claim.response.paymentIntention.paymentOption === domain.PaymentOption.INSTALMENTS %}
+            {{
+              repaymentPlanBlock(
+                heading = '',
+                repaymentPlan = claim.response.paymentIntention.repaymentPlan,
+                paymentPlan = paymentPlan
+              )
+            }}
+          {% elseif claim.response.paymentIntention.paymentOption === domain.PaymentOption.BY_SPECIFIED_DATE %}
+            {{
+              paymentIntentionFragment(
+                amount = claim.response.amount,
+                paymentIntention = claim.response.paymentIntention
+              )
+            }}
+          {% endif %}
+
+          {{ paymentDateExplanationBlock(claim.response.statementOfMeans.reason) }}
+
+          {{ financialDetailsBlock(totalMonthlyIncome, totalMonthlyExpenses, claim.response.statementOfMeans) }}
+        {% endif %}
+      {% endif %}
+
+      <form novalidate method="post">
+        {{ csrfProtection(csrf) }}
+
+        {% if claim.response.responseType === ResponseType.PART_ADMISSION.value and page !== 1 %}
+          <input type="hidden" name="action[showPage]" value="{{ page + 1 }}">
```

Comment:
I prefer this routing logic be moved out of view. We can have two pages with two routes and decide accordingly whether we just need to display one or both. Makes it easier to test and will reduce the code complexity. I am still trying to find how this logic is covered in tests.

Reply:
I looked into alternative of having separate route for details but it comes with lots of extra problems like:
 - name for the extra page
 - lots of additional code that we usually put in routes
 - necessity of sharing view macros between two pages
 - necessity of writing additional tests for extra route

In the meantime I added route test case for pagination.